src/tests/cmocka/test_krb5_common.c

```diff
@@ -0,0 +1,297 @@
+    assert_string_equal(krb5_child_extra_args[1], gid_opt);
+    assert_string_equal(krb5_child_extra_args[2], "--realm=" TEST_REALM);
+    assert_string_equal(krb5_child_extra_args[3], "--canonicalize");
+    assert_null(krb5_child_extra_args[4]);
+    talloc_free(krb5_child_extra_args);
+
+    /* --fast-principal will be only set if FAST is used */
+    krb5_ctx->fast_principal = discard_const(TEST_FAST_PRINC);
+    ret = set_extra_args(global_talloc_context, krb5_ctx,
+                         &krb5_child_extra_args);
+    assert_int_equal(ret, EOK);
+    assert_string_equal(krb5_child_extra_args[0], uid_opt);
+    assert_string_equal(krb5_child_extra_args[1], gid_opt);
+    assert_string_equal(krb5_child_extra_args[2], "--realm=" TEST_REALM);
+    assert_string_equal(krb5_child_extra_args[3], "--canonicalize");
+    assert_null(krb5_child_extra_args[4]);
+    talloc_free(krb5_child_extra_args);
+
+    krb5_ctx->use_fast_str = discard_const(TEST_FAST_STR);
+    ret = set_extra_args(global_talloc_context, krb5_ctx,
+                         &krb5_child_extra_args);
+    assert_int_equal(ret, EOK);
+    assert_string_equal(krb5_child_extra_args[0], uid_opt);
+    assert_string_equal(krb5_child_extra_args[1], gid_opt);
+    assert_string_equal(krb5_child_extra_args[2], "--realm=" TEST_REALM);
+    assert_string_equal(krb5_child_extra_args[3], "--use-fast=" TEST_FAST_STR);
+    assert_string_equal(krb5_child_extra_args[4],
+                        "--fast-principal=" TEST_FAST_PRINC);
+    assert_string_equal(krb5_child_extra_args[5], "--canonicalize");
+    assert_null(krb5_child_extra_args[6]);
+    talloc_free(krb5_child_extra_args);
+
+    krb5_ctx->lifetime_str = discard_const(TEST_LIFE_STR);
+    krb5_ctx->rlife_str = discard_const(TEST_RLIFE_STR);
+    ret = set_extra_args(global_talloc_context, krb5_ctx,
+                         &krb5_child_extra_args);
+    assert_int_equal(ret, EOK);
+    assert_string_equal(krb5_child_extra_args[0], uid_opt);
+    assert_string_equal(krb5_child_extra_args[1], gid_opt);
+    assert_string_equal(krb5_child_extra_args[2], "--realm=" TEST_REALM);
+    assert_string_equal(krb5_child_extra_args[3], "--lifetime=" TEST_LIFE_STR);
+    assert_string_equal(krb5_child_extra_args[4],
+                        "--renewable-lifetime=" TEST_RLIFE_STR);
+    assert_string_equal(krb5_child_extra_args[5], "--use-fast=" TEST_FAST_STR);
+    assert_string_equal(krb5_child_extra_args[6],
+                        "--fast-principal=" TEST_FAST_PRINC);
+    assert_string_equal(krb5_child_extra_args[7], "--canonicalize");
+    assert_null(krb5_child_extra_args[8]);
+    talloc_free(krb5_child_extra_args);
+
+    talloc_free(krb5_ctx);
+}
+
+void test_krb5_check_options(void **state)
+{
+    int ret;
+    struct dp_option *opts;
+    struct test_ctx *test_ctx = talloc_get_type(*state, struct test_ctx);
+    struct krb5_ctx *krb5_ctx;
+
+    ret = krb5_check_options(NULL, NULL, NULL);
```

Comment:
Could you rename the tested function.? We might use prefix `sss_krb5_` or something else
but we should try to avoid possible conflicts with official krb5 functions